### PR TITLE
feat(cicd): add conditional field in pipeline manifest for connection_name

### LINKED ISSUE #1247: Feature: Github Enterprise Server in Pipelines
I'd love the ability to trigger a Copilot Pipeline from a self-hosted Github Enterprise instance.
I see that [CodeBuild already have support GH Enterprise Server](https://docs.aws.amazon.com/codebuild/latest/userguide/sample-github-enterprise.html) but I'm not sure if it's supported by CodePipeline.

### LINKED ISSUE #2012: Is is possible to specify a currently existing CodeStar connection ARN when using Bitbucket as the Source?
Currently the pipeline init flow seems to be setup so that a new CodeStar connection is created. It would be useful to be able to use a pre existing connection (pass the ARN as a flag?) as the Bitbucket authentication process requires a level of privilege that is not available within our org by default, and we have pre-existing connections already setup.

### PR DESCRIPTION
**Pull request by huanjani**
These changes add a commented-out optional field in the pipeline manifest for `connection_name` when the source provider is GitHub or Bitbucket.
Manually tested with non-CSC, Bb, GH, CSC w/ existing connection, CSC w/o existing connection.

Updated docs accordingly.

Related https://github.com/aws/copilot-cli/pull/2042, Resolves #2012, Resolves #1247, Related #1339.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by huanjani** [comment]
Non-CSC:
```
  # Additional properties that further specifies the exact location
  # the artifacts should be sourced from. For example, the GitHub provider
  # has the following properties: repository, branch.
  properties:
    branch: master
    repository: https://us-west-2.console.aws.amazon.com/codesuite/codecommit/repositories/aws-sample/browse

# The deployment section defines the order the pipeline will deploy
# to your environments.
```
CSC:
```
 # Additional properties that further specifies the exact location
  # the artifacts should be sourced from. For example, the GitHub provider
  # has the following properties: repository, branch.
  properties:
    branch: master
    repository: https://huanjani@bitbucket.org/huanjani/aws-copilot-sample-service
    # Optional: specify the name of an existing CodeStar Connections connection.
    # connection_name:

# The deployment section defines the order the pipeline will deploy
# to your environments.
```

File: site/content/docs/manifest/pipeline.md [hunk 1]
```diff
@@ -11,6 +11,8 @@ List of all available properties for a Copilot pipeline manifest.
       properties:
         branch: main
         repository: https://github.com/<user>/sample-app-frontend
+        # Optional: specify the name of an existing CodeStar Connections connection.
+        connection_name:
```
Comment (bvtujo) [inline site/content/docs/manifest/pipeline.md]: Should there be an example connection name here?

Comment (huanjani) [reply site/content/docs/manifest/pipeline.md]: Added. Thanks!

File: internal/pkg/manifest/pipeline.go [hunk 2]
```diff
@@ -230,6 +230,14 @@ func UnmarshalPipeline(in []byte) (*PipelineManifest, error) {
 	return nil, errors.New("unexpected error occurs while unmarshalling pipeline.yml")
 }
 
+// IsCodeStarConnection indicates to the manifest if this source requires a CSC connection.
+func (s Source) IsCodeStarConnection() bool {
+	if s.ProviderName == "GitHub" || s.ProviderName == "Bitbucket" {
```
Comment (efekarakus) [inline internal/pkg/manifest/pipeline.go]: nit:
```go
switch s.ProviderName {
case GithubProviderName:
  return true
case BitbucketProviderName:
  return true
default:
  return false
}
```

File: site/content/docs/concepts/pipelines.md [hunk 3]
```diff
@@ -74,6 +74,8 @@ source:
   properties:
     branch: main
     repository: https://github.com/kohidave/demo-api-frontend
+    # Optional: specify the name of an existing CodeStar Connections connection.
+    # connection_name:
```
Comment (efekarakus) [inline site/content/docs/concepts/pipelines.md]: can we add a sample name here too:
```suggestion
    # connection_name: a-connection
```
Taken from https://docs.aws.amazon.com/dtconsole/latest/userguide/connections-create-bitbucket.html

File: site/content/docs/manifest/pipeline.md [hunk 4]
```diff
@@ -53,6 +55,9 @@ The name of the branch in your repository that triggers the pipeline. The defaul
 <span class="parent-field">source.properties.</span><a id="source-properties-repository" href="#source-properties-repository" class="field">`repository`</a> <span class="type">String</span>  
 The URL of your repository.
 
+<span class="parent-field">source.properties.</span><a id="source-properties-connection-name" href="#source-properties-connection-name" class="field">`connection_name`</a> <span class="type">String</span>  
+The name of an existing CodeStar Connections connection. By default, Copilot will generate a connection for you.
```
Comment (efekarakus) [inline site/content/docs/manifest/pipeline.md]:
```suggestion
The name of an existing CodeStar Connections connection. If omitted, Copilot will generate a connection for you.
```

File: site/content/docs/manifest/pipeline.md [hunk 5]
```diff
@@ -11,6 +11,8 @@ List of all available properties for a Copilot pipeline manifest.
       properties:
         branch: main
         repository: https://github.com/<user>/sample-app-frontend
+        # Optional: specify the name of an existing CodeStar Connections connection.
+        connection_name: MyPreexistingConnection
```
Comment (efekarakus) [inline site/content/docs/manifest/pipeline.md]: nit: samesies here

File: templates/cicd/pipeline.yml [hunk 6]
```diff
@@ -15,6 +15,10 @@ source:
   # has the following properties: repository, branch.
   properties:{{range $key, $value := .Source.Properties}}
     {{$key}}: {{$value}}{{end}}
+    {{- if .Source.IsCodeStarConnection}}
+    # Optional: specify the name of an existing CodeStar Connections connection.
+    # connection_name:
```
Comment (efekarakus) [inline templates/cicd/pipeline.yml]: can we also put a sample name here instead of leaving it blank: 
```
# connection_name: a-connection
```

**Review by efekarakus — COMMENTED** [review]
Code looks good, just some documentation updates 🙏

**Review by bvtujo — APPROVED** [review]
Looks good!

File: site/content/docs/concepts/pipelines.md (hunk 3 above)
Comment (huanjani) [reply site/content/docs/concepts/pipelines.md]: If there's an example, then I should uncomment it, correct?

File: site/content/docs/manifest/pipeline.md (hunk 4 above)
Comment (huanjani) [reply site/content/docs/manifest/pipeline.md]: Ah, that's much better. Thanks!

File: site/content/docs/concepts/pipelines.md (hunk 3 above)
Comment (huanjani) [reply site/content/docs/concepts/pipelines.md]: Nevermind. Just got to your template comment below.

**Review by efekarakus — APPROVED** [review]

**Pull request merged**